source/extensions/common/dynamic_forward_proxy/dns_cache_resource_manager.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <string>
+
+#include "envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.pb.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+#include "envoy/upstream/resource_manager.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace DynamicForwardProxy {
+
+#define ALL_DNS_CACHE_CIRCUIT_BREAKERS_STATS(OPEN_GAUGE, REMAINING_GAUGE)                          \
+  OPEN_GAUGE(rq_pending_opening, Accumulate)                                                       \
+  REMAINING_GAUGE(rq_pending_remaining, Accumulate)
+
+struct DnsCacheCircuitBreakersStats {
+  ALL_DNS_CACHE_CIRCUIT_BREAKERS_STATS(GENERATE_GAUGE_STRUCT, GENERATE_GAUGE_STRUCT)
+};
+
+class DnsResource : public Envoy::Upstream::Resource {
```

Comment:
This is mostly copied from other resource classes. @tonya11en I know you did some work around this recently in the private repo. Is it possible to upstream just that part of the change so it can be used here?

Reply:
Yeah, I'll knock that out today

Reply:
https://github.com/envoyproxy/envoy/pull/11182